renku/core/dataset/providers/api.py

```diff
@@ -143,6 +158,64 @@ def supports_storage(uri: str) -> bool:
         """Whether or not this provider supports a given URI storage."""
         raise NotImplementedError
 
+    def update_files(
+        self,
+        files: List[DynamicProxy],
+        dry_run: bool,
+        delete: bool,
+        context: Dict[str, Any],
+        **kwargs,
+    ) -> List["DatasetUpdateMetadata"]:
+        """Update dataset files from the remote provider."""
+        from renku.core.dataset.providers.models import DatasetUpdateAction, DatasetUpdateMetadata
+
+        progress_text = f"Checking remote files for updates in dataset {files[0].dataset.name}"
+
+        results: List[DatasetUpdateMetadata] = []
+
+        try:
+            communication.start_progress(progress_text, len(files))
+
+            storage = self.get_storage()
+            hashes = storage.get_hashes(uri=files[0].dataset.storage)
+            for file in files:
```

Comment:
> It doesn't generate any results when dry_run is True.

are you sure? because it still appends an entry to the results, just doesn't perform the action.

Reply:
My bad. You're right; I've missed that part.